cmd/fyne/internal/util/mobile.go

```diff
@@ -1,8 +1,57 @@
 package util
 
-import "strings"
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// AndroidBuildToolsPath tries to find the location of the "build-tools" directory.
+// This depends on ANDROID_HOME variable, so you should call RequireAndroidSDK() first.
+func AndroidBuildToolsPath() string {
+	env := os.Getenv("ANDROID_HOME")
+	dir := filepath.Join(env, "build-tools")
+
+	// this may contain a version subdir
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return dir
+	}
+
+	childDir := ""
+	for _, f := range files {
+		if f.Name() == "zipalign" { // no version subdir
+			return dir
+		}
+		if f.IsDir() && childDir == "" {
+			childDir = f.Name()
+		}
+	}
+
+	if childDir == "" {
+		return dir
+	}
+	return filepath.Join(dir, childDir)
+}
+
+// IsAndroid returns true if the given os parameter represents one of the Android targets.
+func IsAndroid(os string) bool {
+	return strings.Index(os, "android") == 0
+}
 
 // IsMobile returns true if the given os parameter represents a platform handled by gomobile.
 func IsMobile(os string) bool {
 	return os == "ios" || strings.HasPrefix(os, "android")
 }
+
+// RequireAndroidSDK will return an error if it cannot establish the location of a valid Android SDK installation.
+// This is currently deduced using ANDROID_HOME environment variable.
+func RequireAndroidSDK() error {
+	if env, ok := os.LookupEnv("ANDROID_HOME"); !ok || env == "" {
```

Comment:
Should this also check `env` points to a readable directory?

Reply:
It's not something we've done elsewhere - when using os.UserHome() we don't check that is a readable directory

Reply:
ack